lib/ddtrace/opentracer/distributed_headers.rb

```diff
@@ -0,0 +1,42 @@
+require 'ddtrace/span'
+require 'ddtrace/ext/distributed'
+
+module Datadog
+  module OpenTracer
+    # DistributedHeaders provides easy access and validation to headers
+    class DistributedHeaders
+      include Datadog::Ext::DistributedTracing
+
+      def initialize(carrier)
+        @carrier = carrier
+      end
+
+      def valid?
+        # Sampling priority is optional.
+        !trace_id.nil? && !parent_id.nil?
+      end
+
+      def trace_id
+        value = @carrier[HTTP_HEADER_TRACE_ID].to_i
+        return if value <= 0 || value >= Datadog::Span::MAX_ID
```

Comment:
Note to self: this doesn't support signed ints. We should apply the changes from #530 here as well, to make sure this works with OpenTracing.